### ISSUE TITLE
Workday settings dialog has no labels to indicate the purpose of the form controls

### ISSUE THREAD

**Issue by andkit** [home-assistant/frontend#18520]
### Checklist

- [X] I have updated to the latest available Home Assistant version.
- [X] I have cleared the cache of my browser.
- [X] I have tried a different browser to see if it is related to my browser.

### Describe the issue you are experiencing

After updating my installation I got an error log telling me that the workday integration is no longer configured via yaml.
So I removed it there and went to check the settings to see that everything is set up correctly.
However I find that dialog supremely unhelpful as none of the controls have a label

I see a couple of days as buttons, an empty drop-down, a number input some more day-buttons, some more empty drop-downs and finally a drop-down with "TH" selected (that is the code for my province)

I looked into .storage/core.config_entries  and there everything looks plausible at first glance.

### Describe the behavior you expected

Labels on the controls, maybe a heading or two so I know what I am actually configuring here.
Also while I can see a control that I *guess* selects the province, I can't see the selected country anywhere

### Steps to reproduce the issue

Open Settings -> Integrations -> Workday -> Configure

### What version of Home Assistant Core has the issue?

 Core 2023.11.0 Frontend 20231030.1

### In which browser are you experiencing the issue with?

Firefox 119

**Comment by andkit** [comment]
![unhelpful_settings](https://github.com/home-assistant/frontend/assets/1395811/f16d59b2-42fe-4ce1-9b56-5260b045f044)

Here is a screenshot of said dialog

**Comment by karwosts** [comment]
Looks fine to me 🤷 

If this happened just after you updated, maybe just try clearing your caches, that sort of thing.

![image](https://github.com/home-assistant/frontend/assets/32912880/25c42172-538b-4665-a27c-82e84dfc8c66)

**Comment by andkit** [comment]
Yes that looks much much better... Hmm can't be browser cache because some browsers I tried never saw home assistant before, but knowing it **shouldn't** look like this I'll play around some more. Maybe I'll try with a clean install, though I probably won't get around to that before the weekend.

Thanks for your feedback
